source/change-streams/tests/unified/change-streams.yml

```diff
@@ -177,3 +217,585 @@ tests:
+      - minServerVersion: 3.6.0
+        topologies:
+          - replicaset
+    operations:
+      - name: createChangeStream
+        object: *collection0
+        arguments:
+          pipeline: []
+        saveResultAsEntity: &changeStream0 changeStream0
+      - name: insertOne
+        object: *globalCollection0
+        arguments:
+          document:
+            x: 1
+      - name: updateOne
+        object: *globalCollection0
+        arguments:
+          filter:
+            x: 1
+          update:
+            $set:
+              x: 2
+      - name: replaceOne
+        object: *globalCollection0
+        arguments:
+          filter:
+            x: 2
+          replacement:
+            x: 3
+      - name: deleteOne
+        object: *globalCollection0
+        arguments:
+          filter:
+            x: 3
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult:
+          operationType: insert
+          ns:
+            db: *database0
+            coll: *collection0
+          fullDocument:
+            x:
+              $numberInt: "1"
+            _id:
+              $$exists: true
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult:
+          operationType: update
+          ns:
+            db: *database0
+            coll: *collection0
+          updateDescription:
+            updatedFields:
+              x:
+                $numberInt: "2"
+            removedFields: []
+            truncatedArrays:
+              $$unsetOrMatches:
+                $$exists:
+                  true
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult:
+          operationType: replace
+          ns:
+            db: *database0
+            coll: *collection0
+          fullDocument:
+            x:
+              $numberInt: "3"
+            _id:
+              $$exists: true
+      - name: iterateUntilDocumentOrError
+        object: *changeStream0
+        expectResult:
+          operationType: delete
+          ns:
+            db: *database0
+            coll: *collection0
+    expectEvents:
+      - client: *client0
+        ignoreExtraEvents: true
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0
+                cursor: {}
+                pipeline:
+                  - $changeStream: {}
+              commandName: aggregate
+              databaseName: *database0
+  - description: Test rename and invalidate event types
+    runOnRequirements:
+      - minServerVersion: 4.0.1
+        topologies:
+          - replicaset
+    operations:
+      - name: createChangeStream
+        object: *collection0
+        arguments:
+          pipeline: []
+        saveResultAsEntity: &changeStream0 changeStream0
+      - name: dropCollection
+        object: *database0
+        arguments:
+          collection: *collection1
+      - name: rename
```

Comment:
I'm not sure this function exists in the unified test format yet. You might need to define it in the spec and bump the version again.

Reply:
Relevant spec section would be [Database Operations](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#database-operations), which states:

> Database operations that require special handling or are not documented by an existing specification are described below.

This is a bit tricky as there is nothing special about the rename operation (with respect to the test runner), other than the fact that we don't have an existing spec for it. I expect this may come up down the line when the retryable reads spec tests get ported, since those have a number of other operations not explicitly defined in other specs (e.g. `mapReduce`, which may very well be obsolete by the time we get to that task).

Either way, I think it makes sense to document this in the unified spec for the time being, so we can at least document the expected arguments. Are there any other special operations we need to handle? I'm not sure if other change stream tests also performed collection-level renames, but if so we should add that as well.

Reply:
I'm pretty sure there aren't any other special operations; `renameCollection` was the only one added to the Rust runner to support these (and apologies that it didn't occur to me when doing so that it would need a spec update), and it looks like that's the only test that uses it.

I'll put together a PR to add `rename` to the unified spec.  Given the above, should it be a version increment to 1.8, rolled into 1.7, or a doc update without a version bump?

Reply:
> should it be a version increment to 1.8, rolled into 1.7, or a doc update without a version bump?

As much as I hate the idea, this is a technically an additive change and warrants a minor version bump. This will probably be the first time we have a minor version without a corresponding schema.

Alternatively, you can float the idea in `#driver-devs` about getting rid of the "spec version" and just relying on schema version. That will require some significant language changes to [Impact of Spec Changes on Schema Version](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#impact-of-spec-changes-on-schema-version), though.

Reply:
I alluded to this discussion in [DRIVERS-2253](https://jira.mongodb.org/browse/DRIVERS-2253).

Reply:
Let's not move forward on [DRIVERS-2253](https://jira.mongodb.org/browse/DRIVERS-2253) until it gets properly scheduled. But in the mean time, I think I can rules-lawyer our way out of requiring a schema version bump.

Quoting [Executing an Operation](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#executing-an-operation):

< If operation.object is not "testRunner", this is an entity operation. If operation.object is defined in the current test's Entity Map, the test runner MUST fetch that entity and note its type; otherwise, the test runner MUST raise an error for an undefined entity. _If operation.name does not correspond to a known operation for the entity type (per Entity Test Operations), the test runner MUST raise an error for an unsupported operation_. Test runners MAY skip tests that include operations that are intentionally unimplemented (e.g. listCollectionNames).

I think this leaves two possibilities:

 * Drivers have implemented the unified test runner correctly and will see an obvious error if they encounter a new test with a `rename` operation and implement it then when syncing the tests. This can also be noted in the downstream changes for DRIVERS-1713.
 * Drivers didn't implement the unified test runner correctly and won't realize the operation is unsupported (e.g. it could be silently ignored). In that case, I'd expect these change stream tests to fail since there wouldn't be a `rename` event returned. Either way, drivers would know to address this.

Feel free to float the idea in #driver-devs. But if anyone is uncomfortable with this approach (starting with @benjirewis), then I have no qualms about just bumping to 1.8 and calling it a day.

If so, just be sure to make a copy of `schema-1.7.json` and commit it as `schema-1.8.json` in this PR. I'd also suggest using `schemaVersion: "1.8"` in _only_ the test files that use `rename` (others can use 1.7), since that might also let test runner implementations catch that _something_ is different from 1.7, even if the JSON schema itself is unchanged. And we also get to avoid having a missing schema file, which would no doubt raise questions and require redundant explanations down the line.

Reply:
@abr-egn: I just saw https://github.com/mongodb/specifications/pull/1138/files#r839991713. Since that's already merged, we can argue the damage is already done (speaking lightheartedly here) and just ignore a schema bump for `rename`. When drivers implement their downstream tickets for [DRIVERS-1343](https://jira.mongodb.org/browse/DRIVERS-1343) they will surely encounter one of the "two possibilities" I alluded to above.

@benjirewis: Do you concur?

Reply:
Yeah I should've seen that in DRIVERS-1343 🤦 ... I think not bumping the schema version to 1.8 (and remaining on 1.7) here is fine. I worry more about the drivers that silently ignore unrecognized operations (I hope this is not many, as that would be an incorrect implementation of the UTR). I'm curious about the followup in DRIVERS-2253 and am all for dropping spec versions beyond the unified test format (as well as other, trivial [metadata](https://jira.mongodb.org/browse/DRIVERS-2253?focusedCommentId=4447139&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4447139)).